src/EasyAbp.Abp.EventBus.Boxes.Dtm.Http/EasyAbp/Abp/EventBus/Boxes/Dtm/HttpDtmMessageManager.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dtmcli;
+using EasyAbp.Abp.EventBus.Boxes.Dtm.Http.EasyAbp.Abp.EventBus.Boxes.Options;
+using EasyAbp.Abp.EventBus.Boxes.Dtm.Models;
+using EasyAbp.Abp.EventBus.Boxes.Dtm.Options;
+using JetBrains.Annotations;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Volo.Abp;
+using Volo.Abp.Data;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.EventBus.Distributed;
+using Volo.Abp.MultiTenancy;
+using Volo.Abp.Uow;
+
+namespace EasyAbp.Abp.EventBus.Boxes.Dtm;
+
+/// <summary>
+/// http 
+/// </summary>
+public class HttpDtmMessageManager : IDtmMessageManager, ITransientDependency
```

Comment:
Can you make a new abstract class named `DtmMessageManagerBase` to share code with the HTTP and gRPC implementation?

Don't remove my comments like:
```c#
        // WARNING:
        // the defaultMsg (with non-transactional events) will not invoke DTM's Prepare and not create a barrier.
        // That means when the DTM crashes, non-transactional events will never publish.
        // To avoid this problem, please KEEP USING TRANSACTION if you need write-operations.
```